extensions/amp-story/1.0/progress-bar.js

```diff
@@ -155,11 +155,14 @@ export class ProgressBar {
         }
 
         this.segmentsAddedPromise_ = this.mutator_.mutateElement(root, () => {
-          /** @type {!Array} */ (pageIds).forEach((id) => {
-            if (!(id in this.segmentIdMap_)) {
-              this.addSegment_(id);
-            }
-          });
+          /** @type {!Array} */ (pageIds)
+            // Do not show progress bar for the ad page.
+            .filter((id) => !id.startsWith('i-amphtml-ad-'))
```

Comment:
Would this not add into the progress bar the section for the ad pages? That's exactly what we want to show. You'd also need to update `buildSegmentEl_`

Reply:
as a first step we don't show any UI change:
1. Refactoring
2. UI change